runtime/src/anchor.rs

```diff
@@ -459,12 +512,31 @@ mod tests {
                 Origin::signed(1),
                 pre_image,
                 doc_root,
-                <Test as system::Trait>::Hashing::hash_of(&0)
+                <Test as system::Trait>::Hashing::hash_of(&0),
+                1567589834087
             ));
             // asserting that the stored anchor id is what we sent the pre-image for
-            let a = Anchor::get_anchor(anchor_id);
+            let mut a = Anchor::get_anchor_by_id(anchor_id).unwrap();
             assert_eq!(a.id, anchor_id);
             assert_eq!(a.doc_root, doc_root);
+            assert_eq!(Anchor::get_anchor_evict_date(anchor_id), 18144);
+            assert_eq!(Anchor::get_anchor_id_by_index(Anchor::get_current_index()), anchor_id);
+            assert_eq!(Anchor::get_anchor_id_by_index(1), anchor_id);
+
+            // commit second anchor to test index updates
+            assert_ok!(Anchor::commit(
+                Origin::signed(1),
+                pre_image2,
+                doc_root,
+                <Test as system::Trait>::Hashing::hash_of(&0),
+                1567589844087
+            ));
+            a = Anchor::get_anchor_by_id(anchor_id2).unwrap();
+            assert_eq!(a.id, anchor_id2);
+            assert_eq!(a.doc_root, doc_root);
+            assert_eq!(Anchor::get_anchor_evict_date(anchor_id2), 18144);
+            assert_eq!(Anchor::get_anchor_id_by_index(2), anchor_id2);
+            assert_eq!(Anchor::get_anchor_id_by_index(Anchor::get_current_index()), anchor_id2);
```

Comment:
if testing the index update, wouldn't it be better to pass the index `2` in here instead of get_current_index? Because what if the logic inside the code was broken and the index would always be stuck to 1? then the second get would still work but still be broken in terms of logic.

Reply:
If I understand correctly the test case you suggest is the one above this line? :)